controlplane/pkg/plugins/utils.go

```diff
@@ -97,3 +101,26 @@ func registerPlugin(name, endpoint, registry string, capabilities []plugins.Plug
 
 	return err
 }
+
+func monitorNSMD(name string, endpoint string, registry string, capabilities []plugins.PluginCapability) {
```

Comment:
please, consider to abstact the monitoring staff and to put it into probes? I think it could be reuseful. For example
probes.StartMonitoring(..., reconnect func() error)

Reply:
in addition, it will be possible to cover unit tests

Reply:
I don't think that adding new abstraction is good idea while we have only one usage of it. We can do it in further refactoring in case of facing logical duplications